source/Platforms.hx

```diff
@@ -0,0 +1,178 @@
+package;
+
+import flixel.FlxG;
+import flixel.FlxObject;
+import flixel.FlxSprite;
+import flixel.system.FlxAssets.FlxGraphicAsset;
+import flixel.util.FlxColor;
+import flixel.input.keyboard.FlxKey;
+import flixel.group.FlxGroup;
+
+/**
+ * ...
+ * @author Jess Geiger
+ */
+class Platforms extends FlxSprite 
+{
+    //set initial movement
+    private var UP:Bool = false;
+    private var DOWN:Bool = true;
+    private var LEFT:Bool = false;
+    private var RIGHT: Bool = true;
+    private var heavy:Float = 8000; // Declare higher gravity while on platform
+    private var blockSize:Int = 32; //Declare block size to be used in width and height
+    //Initialize min and max positioning
+    private var minX:Float;
+    private var maxX:Float;
+    private var minY:Float;
+    private var maxY:Float;
+    //Set starting position and width of platform
+    private var startX:Float;
+    private var startY:Float;
+    private var platformwidth:Int;
+    public var offsetX:Float = 0;
+    public var inContact:Bool = false;
+    private var platformVelocity:Int = 40;
+    public var touchingSprites:FlxTypedGroup<PlatformTracker> = new FlxTypedGroup<PlatformTracker>();
+    private var countTouching:Int = 0;
+
+    public function new(?X:Float=0, ?Y:Float=0, ?W:Int=0, ?L:Float=0, ?R:Float=0, ?U:Float=0, ?D:Float=0, ?trackPlayer:Player, ?SimpleGraphic:FlxGraphicAsset) 
+	{
+		super(X, Y, SimpleGraphic);
+		//set starting positions based on requested dimensions/positions
+        startX = X;
+        startY = Y;
+        minX = L;
+        maxX = R;
+        minY = U;
+        maxY = D;
+        platformwidth = blockSize * W;
+
+		// Initializes a basic graphic for the player
+		makeGraphic(platformwidth, blockSize, FlxColor.YELLOW);
+
+	}
+
+    //Movement as requested by user parameters
+    public function movement():Void
```

Comment:
/**
     *  Movement as requested by user parameters. When the object hits the maximum position in 
     *  one direction, it is directed in the other direction. Corresponding booleans are changed
     *  as well.
     *  
     */